bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/WriteAdvHandle.java

```diff
@@ -45,12 +45,36 @@
      *
      * @param entryId entryId to be added
      * @param data array of bytes to be written
-     * @return an handle to the result, in case of success it will return the same value of param entryId
+     * @return an handle to the result, in case of success it will return the same value of param entryId.
      */
     default CompletableFuture<Long> write(final long entryId, final ByteBuffer data) {
         return write(entryId, Unpooled.wrappedBuffer(data));
     }
 
+    /**
+     * Add entry asynchronously to an open ledger.
+     *
+     * @param entryId entryId to be added.
+     * @param data array of bytes to be written
```

Comment:
should we tell that the array will be processed async and so the caller MUST not chnge the contents of the array ?
It is not so obvious, a user could think that data is copied or is not used anymore once the method returns.
Users of BK are advanced developers, but a line of warning maybe would be good

Reply:
@eolivelli I am not sure if we need that. that basically means we need to add same comment for every methods that take input from users. it applies almost every methods. If you think we need that, let's do it in a separate task and add comments to all the methods.

Reply:
@sijie no problem, just asking your opinion. already gave my +1 on this patch

Reply:
created #757 for future improvements.